environment.yml

```diff
@@ -40,7 +40,7 @@ dependencies:
   - mesalib
   - mysql
   - mysql-connector-c
-  - nanoflann
+  - nanoflann<1.4
```

Comment:
This also needs to be updated in the meta.yml recipe. I don't think we need to worry about the run pin, but the build pin needs to be updated.